apps/designer/app/canvas/features/text-editor/text-editor.tsx

```diff
@@ -54,20 +119,24 @@ export const TextEditor = ({
   contentEditable,
   onChange,
 }: TextEditorProps) => {
+  const [paragraphClassName] = useState(() => nanoid());
   const [italicClassName] = useState(() => nanoid());
 
-  // initially instance styles are applied to all nodes
-  // so not necessary to use layout effect
-  useEffect(() => {
+  useLayoutEffect(() => {
     const engine = createCssEngine();
+    // reset paragraph styles and make it work inside <a>
+    engine.addPlaintextRule(`
```

Comment:
didn't you just removed paragraphs?

Reply:
oh that's for our paragraph instance

Reply:
no its not, ok it seems like you are keeping the lexical paragraph and only replacing pagraphs when new line is hit

Reply:
Yep, one paragraph is lexical requirement. Even in plain text editor.